Calibration/HcalCalibAlgos/macros/CalibFitPlots.C

```diff
@@ -2492,21 +2493,23 @@ void PlotHistCorrAsymmetry(char* infile, std::string text, std::string prefixF =
   std::vector<int> entries;
   char name[100];
   double dy(0);
-  for (int j = 0; j < maxdepth; ++j) {
-    sprintf(name, "hd%d", j + 1);
+  int maxd = (depth < 0) ? maxdepth : 1;
+  for (int j = 0; j < maxd; ++j) {
+    int dep = (depth < 0) ? (j + 1) : depth;
```

Comment:
[L2526-2528](https://github.com/cms-sw/cmssw/pull/47042/files#diff-bb14a11f3aafaa7cdad2223c369104686aac440729ff33ace1c1f76ebc36a906R2526-R2528) below will not work if `depth = 0` . Probably that value of the parameter will never be passed, but with the following update such a crash can be avoided:
```suggestion
    int dep = (depth <= 0) ? (j + 1) : depth;
```
Altrernatively, one could set here [above](https://github.com/cms-sw/cmssw/pull/47042/files#diff-bb14a11f3aafaa7cdad2223c369104686aac440729ff33ace1c1f76ebc36a906R2496) `maxd = 0` if `depth = 0`

Reply:
Or (probably even simpler) use "0" as default value for `depth`, rather than "-1"